src/components/intersection.js

```diff
@@ -0,0 +1,353 @@
+/* global AFRAME */
+AFRAME.registerComponent('intersection', {
+  schema: {
+    dimensions: { type: 'string', default: '20 20' },
+    sidewalk: { type: 'string', default: '0 0 0 0' },
+    northeastcurb: { type: 'string', default: '0 0' },
+    southwestcurb: { type: 'string', default: '0 0' },
+    southeastcurb: { type: 'string', default: '0 0' },
+    northwestcurb: { type: 'string', default: '0 0' },
+    stopsign: { type: 'string', default: '0 0 0 0' },
+    trafficsignal: { type: 'string', default: '0 0 0 0' },
+    crosswalk: { type: 'string', default: '0 0 0 0' }
+  },
+  update: function () {
+    var data = this.data;
+    var el = this.el;
+
+    // remove all child nodes if exists
+    while (el.firstChild) {
+      el.removeChild(el.lastChild);
+    }
+    const dimensionsArray = data.dimensions.split(' ').map((i) => Number(i));
+    const positionArray = [
+      this.el.getAttribute('position').x,
+      this.el.getAttribute('position').y,
+      this.el.getAttribute('position').z
+    ];
+    const sidewalkArray = data.sidewalk.split(' ').map((i) => Number(i));
+    const northeastcurbArray = data.northeastcurb
+      .split(' ')
+      .map((i) => Number(i));
+    const southwestcurbArray = data.southwestcurb
+      .split(' ')
+      .map((i) => Number(i));
+    const southeastcurbArray = data.southeastcurb
+      .split(' ')
+      .map((i) => Number(i));
+    const northwestcurbArray = data.northwestcurb
+      .split(' ')
+      .map((i) => Number(i));
+    const stopsignArray = data.stopsign.split(' ').map((i) => Number(i));
+    const trafficsignalArray = data.trafficsignal
+      .split(' ')
+      .map((i) => Number(i));
+    const crosswalklArray = data.crosswalk.split(' ').map((i) => Number(i));
+
+    const intersectWidth = dimensionsArray[0];
+    const intersectDepth = dimensionsArray[1];
+
+    this.el.setAttribute(
+      'geometry',
+      `primitive:box; width: ${intersectWidth}; height: ${intersectDepth}; depth:0.2`
+    );
+    this.el.setAttribute('position', {
+      x: positionArray[0],
+      y: -0.1,
+      z: positionArray[2]
+    });
+    this.el.setAttribute('rotation', '-90 0 0');
+    this.el.setAttribute(
+      'material',
+      'src: #asphalt-texture; repeat:5 5; roughness:1'
+    );
+
+    function createSidewalkElem({
+      length,
+      width,
+      positionVec,
+      scaleVec = { x: 1, y: 1, z: 1 },
+      rotationVec
+    }) {
+      const sd = document.createElement('a-entity');
+      const repeatCountInter = [];
+      repeatCountInter[0] = width / 2;
+      // every 2 meters repeat sidewalk texture
+      repeatCountInter[1] = parseInt(length / 2);
+
+      sd.setAttribute('geometry', 'primitive', 'box');
+      sd.setAttribute('geometry', 'height: 0.4');
```

Comment:
I know you just moved the code in a separate file, but I'm reading this code for the first time, so making suggestions here.
This should be done in one call.
```js
sd.setAttribute('geometry', 'primitive: box; height: 0.4');
```

Reply:
yes I know about this, thanks for reminding